src/Microsoft.ML.TimeSeries/SequentialTransformerBase.cs

```diff
@@ -479,14 +479,13 @@ IDataTransform ITransformTemplate.ApplyToData(IHostEnvironment env, IDataView ne
 
             public override Schema OutputSchema => _bindings.Schema;
 
-            public Func<int, bool> GetDependencies(Func<int, bool> predicate)
+            /// <summary>
+            /// Given a set of columns, return the input columns that are needed to generate those output columns.
+            /// </summary>
+            public IEnumerable<Schema.Column> GetDependencies(IEnumerable<Schema.Column> dependingColumns)
             {
-                for (int i = 0; i < OutputSchema.Count; i++)
-                {
-                    if (predicate(i))
-                        return col => true;
-                }
-                return col => false;
+                var columnNames = dependingColumns.Select(col => col.Name);
+                return InputSchema.Where(col => columnNames.Contains(col.Name));
```

Comment:
>InputSchema [](start = 23, length = 11)

was it mistake in previous code? We used to filter by OutputSchema, now it's InputSchema

Reply:
The description of the function was: 

"Given a predicate specifying which columns are needed, return a predicate indicating which input columns are
 needed. " So i took it as : input is columns from the outputschema, and return value is columns from the input schema. 

i thought the iteration is over the OutputSchema, since the predicate was over the OutputSchema. 

hmm, going back to the IRowToRowMapper, summary: 

" The domain of the function is defined over the indices of the columns of <see cref="Schema.Count"/> for <see cref="InputSchema"/>."

but InputSchema => OutputSchema makes no sense, no?


---
In reply to: [256136896](https://github.com/dotnet/machinelearning/pull/2504#discussion_r256136896) [](ancestors = 256136896)

Reply:
"The domain of the function is defined...", I think this refers to the predicate returned by GetDependencies().

The way I understand the old code is - if any columns are active, then activate all the input columns. If this is correct, then the new code should return all the columns of InputSchema if dependingColumns is not empty, and an empty enumerable if dependingColumns is empty.


---
In reply to: [256140629](https://github.com/dotnet/machinelearning/pull/2504#discussion_r256140629) [](ancestors = 256140629,256136896)

Reply:
thanks @yaeldekel

---
In reply to: [256199343](https://github.com/dotnet/machinelearning/pull/2504#discussion_r256199343) [](ancestors = 256199343,256140629,256136896)